src/content/lib/prefs.js

```diff
@@ -41,16 +41,83 @@ class Prefs {
   }
 
   set(name, value) {
-    return this._store.set(name, value);
+    const oldValue = this.get(name);
+    if (oldValue !== value) {
+      this._store.set(name, value);
+      this._emit(name, oldValue, value);
+    }
   }
 
   reset(name) {
-    return this._store.delete(name);
+    const oldValue = this.get(name);
+    const newValue = this._defaults[name];
+    if (oldValue !== newValue) {
+      this._store.delete(name);
+      this._emit(name, oldValue, newValue);
+    }
```

Comment:
That would permanently store the default value as the user value. Which isn't how default values are supposed to work. Consider for example what happens if we want to change the default value for a pref. But the previous default value has already been saved as a user value. So when something requests the value, it won't get the new default value, it'll get the user value (IOW, the old default value).

It doesn't really matter since I think we don't even use this method, but still not ideal.

Actually, this code as I have it is not quite right, either. I seem to have absentmindedly copied from the set method, but it really should delete the value no matter what. Since as it is right now, it will only delete the value if it differs from the default value. But that means it fails to delete a user value that equals the default value. Which prevents cleaning up unnecessary user values.

Basically we only want to emit an event if this resulted in the "effective value" changing, i.e. if deleting the value would result in `get()` returning a different value. If the user value was already equal to the default value, then resetting won't change the value, even though it _should_ delete the value from storage. So we want to delete from storage but only emit an event if the deleted value was non-default.